core/src/main/java/org/infinispan/remoting/rpc/RpcManager.java

```diff
@@ -20,6 +24,78 @@
+    *
+    * @since 9.2
+    */
+   @Experimental
+   <T> CompletionStage<T> invokeCommand(Address target, ReplicableCommand command,
+                                        ResponseCollector<T> collector, RpcOptions rpcOptions);
+
+   /**
+    * Invoke a command on a collection of node and pass the responses to a {@link ResponseCollector}.
+    *
+    * If one of the targets is the local nodes and the delivery order is not {@link DeliverOrder#TOTAL},
+    * the command is only executed on the remote nodes.
+    *
+    * @since 9.2
+    */
+   @Experimental
+   <T> CompletionStage<T> invokeCommand(Collection<Address> targets, ReplicableCommand command,
+                                        ResponseCollector<T> collector, RpcOptions rpcOptions);
+
+   /**
+    * Invoke a command on all the nodes in the cluster and pass the responses to a {@link ResponseCollector}.
+    *
+    * The command is only executed on the local node if the delivery order is {@link DeliverOrder#TOTAL}.
+    * The command is not sent across RELAY2 bridges to remote sites.
+    *
+    * @since 9.2
+    */
+   @Experimental
+   <T> CompletionStage<T> invokeCommandOnAll(ReplicableCommand command, ResponseCollector<T> collector,
+                                             RpcOptions rpcOptions);
+
+   /**
+    * Invoke a command on a collection of nodes and pass the responses to a {@link ResponseCollector}.
+    *
+    * The command is only sent immediately to the first target, and there is an implementation-dependent
+    * delay before sending the command to each target. There is no delay if the target responds or leaves
+    * the cluster. The remaining targets are skipped if {@link ResponseCollector#addResponse(Address, Response)}
+    * returns a non-{@code null} value.
+    *
+    * If one of the targets is the local node and the delivery order is not {@link DeliverOrder#TOTAL},
+    * the command is only executed on the remote nodes.
+    *
+    * @since 9.2
+    */
+   @Experimental
+   <T> CompletionStage<T> invokeCommandStaggered(Collection<Address> targets, ReplicableCommand command,
+                                                 ResponseCollector<T> collector, RpcOptions rpcOptions);
+
+   /**
+    * Invoke different commands on a collection of nodes and pass the responses to a {@link ResponseCollector}.
+    *
+    * If one of the targets is the local node and the delivery order is not {@link DeliverOrder#TOTAL},
+    * the command is only executed on the remote nodes.
+    *
+    * @since 9.2
+    */
+   @Experimental
+   <T> CompletionStage<T> invokeCommands(Collection<Address> targets,
+                                         Function<Address, ReplicableCommand> commandGenerator,
+                                         ResponseCollector<T> collector, RpcOptions rpcOptions);
+
+   /**
+    * Block on a request and return its result.
+    *
+    * @since 9.2
+    */
+   <T> T blocking(CompletionStage<T> request);
```

Comment:
The reason I put it in `RpcManager` is that the exception re-wrapping seemed RPC-specific. I'll experiment with moving it to `CompletableFutures`, but in the meantime I've marked it as `@Experimental` and I hope to remove all its uses before 9.2.